core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -80,9 +98,26 @@ private ManifestFiles() {
    * @return a {@link ManifestReader}
    */
   public static ManifestReader<DataFile> read(ManifestFile manifest, FileIO io, Map<Integer, PartitionSpec> specsById) {
+    return read(manifest, io, EncryptionManagers.defaultManager(), specsById);
+  }
+
+  /**
+   * Returns a new {@link ManifestReader} for an encrypted {@link ManifestFile}.
+   *
+   * @param manifest an encrypted {@link ManifestFile}
+   * @param io a {@link FileIO}
+   * @param encryption a {@link EncryptionManager}
+   * @param specsById a Map from spec ID to partition spec
+   * @return a {@link ManifestReader}
+   */
+  public static ManifestReader<DataFile> read(ManifestFile manifest, FileIO io, EncryptionManager encryption,
+                                              Map<Integer, PartitionSpec> specsById) {
     Preconditions.checkArgument(manifest.content() == ManifestContent.DATA,
-        "Cannot read a delete manifest with a ManifestReader: %s", manifest);
-    InputFile file = io.newInputFile(manifest.path());
+        "Cannot read a delete manifest with ManifestReader: %s", manifest);
```

Comment:
I'm not sure that the previous version of this error message is wrong since it's trying to create a ManifestReader to read a delete manifest ... I dunno, makes enough sense either way to me although I think I prefer the original

Reply:
sorry that's definitely an unintentional change...